### PR TITLE
Add `ForEach`, `ContinueIf` and `BreakIf` to the UIA operation abstraction library

### LINKED ISSUE #21: Consider adding BreakIf and ContinueIf helpers to the C++ Abstraction library
When writing remote operation loop, consumers will frequently end up with a line such as this in the body of the loop:

```c++
                operationScope.If(ancestorElement.IsNull(), [&]()
                {
                    operationScope.Break();
                });
```

It could be helpful to add a helper that lets consumers write:

```c++
                operationScope.BreakIf(ancestorElement.IsNull());
```

This would be a little bit less verbose.

The same thing could be said for `Continue`...

### LINKED ISSUE #22: Consider adding a ForEach helper to the C++ Abstraction
As in any programming scenario, users often want to write a loop that just iterates over elements of a collection. To do this with the Remote Op abstraction, one can write this:

```c++
            const auto arraySize = array.Size();
            UiaUint elementIndex{ 0 };

            operationScope.For(
                []() {} /* initialize */,
                [&]() { return (elementIndex < arraySize); } /* condition */,
                [&]() { elementIndex += 1; } /* modification */,
                [&]() /* body */
                {
                    const auto element = array.GetAt(elementIndex);
                    /* Now after all this ceremony do something with element */
                });
```

We could add a helper to make this less verbose and far more readable:

```c++
operationScope.ForEach(array, [&](auto element)
{
    /* Do something with element! */
});
```

Under the hood, this could do the same thing as the verbose code to set up a for loop. (Or it could even generate some more efficient bytecode in the future...)

### PR DESCRIPTION
**Pull request by LukaszMakar**
Today, iterating over and working with a collection of elements in a `UiaArray` is a relatively verbose process that requires from the code author to create a full-fledged `For` loop that takes multiple lambda expressions (initialize, break condition, post-loop increment and the body of the loop) and variables (size of the collection, element index and collection element).

To make the process smoother, this change adds 3 proposed (in the linked work items) improvements to the UIA operation abstraction library:
* `ContinueIf` that is an equivalent of an `If` statement that -- if met -- skips over the currently processed loop iteration (`continue`),
* `BreakIf` which is a complementary concept to the first one that -- instead of `continue`-ing past the current loop iteration -- `break`s the loop.
* `ForEach` which mimics known, simplified constructs of iterating over elements in a collection and processing them in the order in which the collection exposes them (requiring only the collection to iterate over and the body of the loop to process collection elements).

The change comes with a test that covers all 3 new constructs as a package of improvements around looping over collections of items.

Fixes #22
Fixes #21

### CODE REVIEW

File: src/UIAutomation/UiaOperationAbstraction/UiaTypeAbstractionImpl.g.cpp [hunk 1]
```diff
@@ -6982,6 +6982,23 @@
         return localPattern;
     }
 
+    UiaElement UiaElement::BuildUpdatedCache(UiaCacheRequest cacheRequest)
```
Comment (LukaszMakar) [inline src/UIAutomation/UiaOperationAbstraction/UiaTypeAbstractionImpl.g.cpp:6985]: How...

File: src/UIAutomation/FunctionalTests/UiaOperationAbstractionTests.cpp [hunk 2]
```diff
@@ -479,5 +479,58 @@ namespace UiaOperationAbstractionTests
...
+            app.Activate();
+
+            // Set focus to the display element.
+            auto focusedElement = WaitForElementFocus(L"Display is 0");
+
+            // Initialize the UIA Remote Operation abstraction.
+            const auto cleanup = InitializeUiaOperationAbstraction(useRemoteOperations);
+
+            {
+                auto operationScope = UiaOperationScope::StartNew();
```
Comment (LukaszMakar) [inline src/UIAutomation/FunctionalTests/UiaOperationAbstractionTests.cpp:496]: If you feel strongly about breaking this up into 2-3 independent tests for more functionality-isolated coverage, I can definitely do it.

**Review by zhayuli — APPROVED** [review]
:shipit:

File: src/UIAutomation/FunctionalTests/UiaOperationAbstractionTests.cpp (hunk 2 above)
Comment (mlalic) [reply src/UIAutomation/FunctionalTests/UiaOperationAbstractionTests.cpp:496]: looks good :)

**Review by mlalic — APPROVED** [review]
🚢

**Pull request merged**